### PR TITLE
Fix Claude OAuth cache_control regressions and gzip error decoding

### LINKED ISSUE #1769: [BUG] claude_executor returns garbled gzip error bodies + does not strip excess cache_control blocks for Amp CLI
**Is it a request payload issue?**
[x] Yes, this is a request payload issue. I am using a client/cURL to send a request payload, but I received an unexpected error.
[  ] No, it's another issue.

**Describe the bug**

Two related issues when using Amp CLI through CLIProxyAPI with Claude OAuth:

### 1. Garbled error responses from `claude_executor.go`

Claude's API returns **gzip-compressed error bodies** (Content-Encoding: gzip), but `claude_executor.go:322` logs/returns the raw compressed bytes instead of decompressing them first. This makes every Claude error completely unreadable — both in logs and in the client.

**Log output (garbled):**
```
[claude_executor.go:322] request error, error status: 400, error message: �      4��
```

**Hex dump of the "error message" confirms gzip:**
```
00000000: 1f8b 0800 0000 0000 0003 348c cd0a ...
           ^^^^ gzip magic number
```

**Actual error after manually decompressing the gzip blob:**
```json
{
  "type": "error",
  "error": {
    "type": "invalid_request_error",
    "message": "A maximum of 4 blocks with cache_control may be provided. Found 8."
  }
}
```

### 2. Amp sends 8-9 `cache_control` blocks, Claude OAuth allows max 4

When Amp CLI sends requests through CLIProxyAPI to Claude OAuth, the request payload contains 8-9 message blocks with `cache_control` breakpoints. Claude's API (via OAuth/Pro) enforces a maximum of 4. The paid `ampcode.com` upstream presumably has a higher limit or strips extras.

CLIProxyAPI should ideally **trim excess `cache_control` blocks** (keeping the last 4) before forwarding to Claude OAuth, since this is a known limit of the Claude API for non-API-key (OAuth) access.

**CLI Type**
Claude OAuth (via `--claude-login`) + Amp CLI integration (`ampcode` config section)

**Model Name**
`claude-opus-4-6`

**LLM Client**
Amp CLI (ampcode.com)

**Request Information**

`request-log: true` was enabled. The routing chain from logs:

```
[fallback_handlers.go:56] amp using local provider for model: claude-opus-4-6 provider=claude
[conductor.go:2280] Use OAuth provider=claude auth_file=claude-elucchesi@villagehandcrafted.com.json for model claude-opus-4-6
[apply.go:164] thinking: no config found, passthrough | provider=claude model=claude-opus-4-6
[claude_executor.go:322] request error, error status: 400, error message: <garbled gzip bytes>
```

This repeats for every retry, then falls back to OpenAI-compatibility (OpenCode Zen) which also fails with 429 quota exceeded. The user sees either a garbled error or "Subscription quota exceeded" — neither indicates the real problem.

**Expected behavior**

1. `claude_executor.go` should decompress gzip response bodies before logging/returning error messages
2. Ideally, CLIProxyAPI should strip excess `cache_control` blocks (down to 4) before forwarding to Claude OAuth, since Amp aggressively tags 8-9 blocks for prompt caching

**OS Type**
- OS: macOS 26.3
- CLIProxyAPI: HEAD-1ae994b (Homebrew)
- Amp CLI: latest

**Additional context**

- All 20+ failed requests in a session were this same error — 100% reproducible
- `claude-haiku-4-5` works fine through the same OAuth token (1 successful request logged)
- The Claude OAuth token is valid and not expired
- Decoded every garbled blob in the log (87 instances) — all identical: `"A maximum of 4 blocks with cache_control may be provided. Found 8."` (or 9)

### PR DESCRIPTION
**Pull request by edlsh**
## Summary

This fixes the Claude OAuth regressions reported in #1769 (Amp CLI / Claude Code 400 `invalid_request_error`) by hardening cache-control handling and error decoding in the Claude executor.

### What changed

- Decode compressed error responses (gzip/deflate/br/zstd via existing decoder) on non-2xx paths in:
  - `Execute`
  - `ExecuteStream`
  - `CountTokens`
- Add cache-control limit guard: `enforceCacheControlLimit(payload, 4)`
  - preserves highest-value breakpoints first
  - strips lower-value breakpoints in deterministic phases
- Add TTL ordering guard: `normalizeCacheControlTTL(payload)`
  - enforces compatibility with `prompt-caching-scope-2026-01-05` by downgrading late `ttl=1h` blocks after a 5m/default block in eval order (`tools -> system -> messages`)
- Cloaking refinement:
  - remove `cache_control` from the injected Claude-agent identifier system block
  - stop forcing `ttl=1h` on injected user system blocks (use default 5m behavior)
- Apply cache-control guards to `count_tokens` requests as well (same request-shape constraints)

## Tests

Added executor coverage for:

- TTL normalization behavior across sections
- cache-control limit enforcement prioritization
- tool-only overflow edge case (still enforces max=4)
- `CountTokens` request-path guardrails

Ran:

- `go test ./internal/runtime/executor` ✅

Note: full `go test ./...` has unrelated pre-existing failures in other packages, not touched by this PR.

Fixes #1769

### CODE REVIEW

File: internal/runtime/executor/claude_executor.go [hunk 1]
```diff
@@ -176,11 +185,18 @@ func (e *ClaudeExecutor) Execute(ctx context.Context, auth *cliproxyauth.Auth, r
...
 	recordAPIResponseMetadata(ctx, e.cfg, httpResp.StatusCode, httpResp.Header.Clone())
 	if httpResp.StatusCode < 200 || httpResp.StatusCode >= 300 {
-		b, _ := io.ReadAll(httpResp.Body)
+		// Decompress error responses (e.g. gzip-compressed 400 errors from Anthropic API)
+		errBody := httpResp.Body
+		if ce := httpResp.Header.Get("Content-Encoding"); ce != "" {
+			if decoded, decErr := decodeResponseBody(httpResp.Body, ce); decErr == nil {
+				errBody = decoded
+			}
+		}
```
[the remark this answers is not in the document]
Comment (edlsh) [reply internal/runtime/executor/claude_executor.go:200]: Good catch — implemented. We now handle decode failures explicitly, return a clear statusErr message, and avoid reading a potentially closed body.

File: internal/runtime/executor/claude_executor.go [hunk 2]
```diff
@@ -317,10 +339,17 @@ func (e *ClaudeExecutor) ExecuteStream(ctx context.Context, auth *cliproxyauth.A
...
```
[the remark this answers is not in the document]
Comment (edlsh) [reply internal/runtime/executor/claude_executor.go:365]: Implemented in the streaming path as well: decode errors are now handled explicitly with a clear error message and early return.

File: internal/runtime/executor/claude_executor.go [hunk 3]
```diff
@@ -464,9 +497,16 @@ func (e *ClaudeExecutor) CountTokens(ctx context.Context, auth *cliproxyauth.Aut
...
 	recordAPIResponseMetadata(ctx, e.cfg, resp.StatusCode, resp.Header.Clone())
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		b, _ := io.ReadAll(resp.Body)
+		// Decompress error responses (e.g. gzip-compressed 400 errors from Anthropic API)
+		errBody := io.ReadCloser(resp.Body)
+		if ce := resp.Header.Get("Content-Encoding"); ce != "" {
+			if decoded, decErr := decodeResponseBody(resp.Body, ce); decErr == nil {
+				errBody = decoded
+			}
+		}
```
[the remark this answers is not in the document]
Comment (edlsh) [reply internal/runtime/executor/claude_executor.go]: Implemented. CountTokens now handles decode failures safely, and the redundant io.ReadCloser(resp.Body) cast has been removed.

File: internal/runtime/executor/claude_executor.go [hunk 4]
```diff
@@ -1254,6 +1299,245 @@ func countCacheControls(payload []byte) int {
...
+				})
+			}
+			msgIdx++
+			return true
+		})
+	}
+
+	return payload
+}
+
```
[the remark this answers is not in the document]
Comment (edlsh) [reply internal/runtime/executor/claude_executor.go]: Implemented now. Refactored cache-control normalization/limit enforcement to parse JSON once, mutate in-memory, and marshal once.

**Review by luispater — APPROVED** [review]
